rfc-120-csv-preview.md

```diff
@@ -0,0 +1,134 @@
+
+Once this is done, we can remove some of the config for [the `assets` domain](https://github.com/alphagov/govuk-puppet/blob/51d7f7b648e8fde5aae3adfb774ec3bca6325bd8/modules/router/templates/assets_origin.conf.erb#L69-L74) that supported the legacy feature ([`gov.uk` routing](https://github.com/alphagov/government-frontend/blob/dcbc563030c470e6242d6432896d84feebb2138b/config/routes.rb#L14) remains). We can also delete the code for the legacy feature.
+
+> We may want to add redirects for the old preview URLs in Whitehall, if they continue to receive lots of traffic after all the 'View Online' links have updated.
+
+
+## Rejected proposals
+
+### Use a top-level prefix for the routing (e.g. `/preview/:base_path/:attachment_id`)
+
+This would involve registering a new `/preview` prefix route in [router][router], which would set the backend (rendering app) to Government Frontend for all paths beginning with that route.
+
+- This involves less effort than individually managing lower-level routes for each attachment preview. However, because we don't have any knowledge of these routes, we can't help the user if the URL they enter is invalid; this could happen when the asset is removed, or the parent document changes e.g. unpublished, redirected.
+
+- Using a top-level prefix for CSV Preview is at-odds with the behaviour of other top-level prefixes (`/info`, `/api/content`), since it only works for specific URLs, and doesn't apply to GOV.UK content in general.
+
+### Improve the feature in Whitehall (frontend)
+
+This would involve rewriting the existing CSV Preview feature so that it can work for documents published elsewhere, instead of being coupled to the attachment model in Whitehall.
+
+> We could make the feature work for attachments published by others apps, using the proposed URL structure, which gives us the parent document and the `asset-id` of attachment being previewed.
+
+- This would mean continued reliance of the Whitehall (frontend) codebase and the `/government` prefix route, which has been identified as [tech debt][government-prefix].
+
+- Not using specific routes for each attachment preview means we won't be able to do think like automatic redirects (if the base path of the parent document changes).
+
+### Using sub-routes of the parent document for the routing
+
+The Publishing API supports any number of `routes` for a content item. Normally, this will just be a single route, equal to the `base_path` of the content item. We could add an extra route or routes to support CSV Preview.
+
+> Any additional routes that exist (and which are not redirects) will use the same rendering app as the 'main' route. This fits with the proposal to re-implement CSV Preview in Government Frontend.
+
+> We could use a prefix sub-route (`.../*`), or specific sub-routes for each attachment. Using a prefix sub-route is less effort, but the behaviour is ambiguous if an attachment is not found or associated with the document.
```

Comment:
I'm finding this a little difficult to parse - could you provide example routes for lines 105-109?

Reply:
Yeah, I see what you mean. The URL structure is unchanged for this proposal; it's the routing mechanism that's in discussion here. I've added a (link to) appendix that shows what doing using each of these mechanisms would look like.

Reply:
Thanks 👍